src/components/DiscordVerification.tsx

```diff
@@ -1,78 +1,385 @@
+    };
+
+    if (isPolling) {
+      setProgress(0);
+      attemptCount.current = 0;
+
+      // verify check
+      checkIntervalRef.current = setInterval(async () => {
+        try {
+          attemptCount.current += 1;
+          const isVerified = await checkVerified();
+
+          if (isVerified) {
+            clearIntervals();
+            setIsPolling(false);
+            setShowSuccess(true);
+            setTimeout(onVerificationSuccess, 1500);
+          } else if (attemptCount.current >= MAX_ATTEMPTS) {
+            giveUp();
+          }
+        } catch (error) {
+          devPrint(error);
+          giveUp();
+        }
+      }, CHECK_INTERVAL);
+
+      // progress animation
+      progressIntervalRef.current = setInterval(() => {
+        setProgress((prev) => {
+          const newProgress = prev + 2;
+          return newProgress >= 100 ? 0 : newProgress;
+        });
+      }, PROGRESS_INTERVAL);
+    }
+
+    return clearIntervals;
+  }, [checkVerified, isPolling, onVerificationSuccess]);
+
+  useEffect(() => {
+    if (verificationFailed) {
+      toast({
+        title: 'Verification Timeout',
+        description: 'Please try running the command again',
+        status: 'error',
+        duration: 5000,
+        isClosable: true,
+        render: ({ onClose }) => (
+          <Alert
+            status="error"
+            variant="solid"
+            borderRadius="md"
+            onClick={onClose}
+            cursor="pointer"
+            bg="red.500"
+          >
+            <AlertIcon />
+            <Box flex="1">
+              <AlertTitle fontWeight="bold" color="white">
+                Verification Timeout
+              </AlertTitle>
+              <AlertDescription color="whiteAlpha.900">
+                Please try running the command again
+              </AlertDescription>
+            </Box>
+          </Alert>
+        ),
+      });
+    }
+  }, [verificationFailed, toast]);
 
   const startChecking = async () => {
     setIsLoading(true);
     setIsPolling(true);
+    setVerificationFailed(false);
+
+    toast({
+      title: 'Checking Verification',
+      description: `Checking status for ${MAX_ATTEMPTS * 5} seconds`,
+      status: 'info',
+      duration: 4000,
+      isClosable: true,
+      render: ({ onClose }) => (
+        <Alert
+          status="info"
+          variant="solid"
+          borderRadius="md"
+          onClick={onClose}
+          cursor="pointer"
+          bg="blue.500"
+        >
+          <AlertIcon />
+          <Box flex="1">
+            <AlertTitle fontWeight="bold" color="white">
+              Checking Verification
+            </AlertTitle>
+            <AlertDescription color="whiteAlpha.900">
+              Checking status for {MAX_ATTEMPTS * 5} seconds
+            </AlertDescription>
+          </Box>
+        </Alert>
+      ),
+    });
+  };
+
+  const copyCommand = () => {
+    navigator.clipboard.writeText(`/verify ${username}`);
```

Comment:
I don't think you can just paste it, also /verify opens a modal in discord IIRC

Reply:
ah rats you're right

Reply:
pasting just /verify and pushing enter makes the command selector pop up so I'll leave the copy in